scripts/expX2.py

```diff
@@ -0,0 +1,763 @@
+# dusage                     List home directory disk usage
+# dusage -p <project>        List project disk usage
+# scontrol show job <id>     Details of a job
+
+
+###   If a job failed due to
+#       perl: warning: Setting locale failed.
+#       perl: warning: Please check that your locale settings:
+#   	    LANGUAGE = (unset),
+#   	    LC_ALL = (unset),
+#   	    LC_CTYPE = "UTF-8",
+#   	    LANG = "en_US.iso885915"
+#       are supported and installed on your system.
+#   you may fix it by executing
+#       export LANG=en_US.UTF-8
+#       export LC_ALL=en_US.UTF-8
+
+import io
+import os
+import pathlib
+import random
+import shutil
+import stat
+import statistics
+
+import math
+import sys
+
+
+
+EXP_ID = "evomaster"
+
+
+
+if len(sys.argv) < 9 or len(sys.argv) > 11:
+    print(
+        "Usage:\n<nameOfScript>.py <cluster> <baseSeed> <dir> <minSeed> <maxSeed> <maxActions> <minutesPerRun> <nJobs> <configFilter?> <sutFilter? eg a,b>")
+    exit(1)
+
+
+### input parameters
+
+# Whether .sh are meant to run on cluster or locally
+CLUSTER = sys.argv[1].lower() in ("yes", "true", "t")
+
+# base seeds used for EM runs. TCP port bindings will be based on such seed.
+# If running new experiments while some previous are still running, to avoid TCP port
+# conflict, can use an higher base seed. Each EM run reserves 10 ports. So, if you run
+# 500 jobs with starting seed 10000, you will end up using ports up to 15000
+BASE_SEED = int(sys.argv[2])
+
+# When creating a new set of experiments, all needed files will be saved in a folder
+BASE_DIR = os.path.abspath(sys.argv[3])
+
+# Experiments are repeated a certain number of times, with different seed for the
+# random generator. This specify the starting seed.
+MIN_SEED = int(sys.argv[4])
+
+# Max seed, included. For example, if running min=10 and max=39, each experiment is
+# going to be repeated 30 times, starting from seed 10 to seed 39 (both included).
+MAX_SEED = int(sys.argv[5])
+
+# By default, experiments are run with stopping criterion of number of actions and not time.
+MAX_ACTIONS = int(sys.argv[6])
+
+# How many minutes we expect each EM run to last AT MOST.
+# Warning: if this value is under-estimated, it will happen the cluster will kill jobs
+# that are not finished withing the specified time.
+MINUTES_PER_RUN = int(sys.argv[7])
+
+# How many scripts M we want the N jobs to be divided into.
+# Note: on cluster we can at most submit 400 scripts.
+# Also not that in the same .sh script there can be experiments only for a single SUT.
+NJOBS = int(sys.argv[8])
+
+# Specify a string to filter CONFIGS to be included
+# None or `all` represents all CONFIGS should be included
+# Default is None
+CONFIGFILTER = None
+if len(sys.argv) > 9:
+    CONFIGFILTER = str(sys.argv[9])
+
+# Specify a string to filter suts to be included based on their names
+# None or `all` represents all Suts should be included
+# Default is None
```

Comment:
specify here as a comment that this can be a `,` separated list of names, like `a,b`